engine/src/function.cpp

```diff
@@ -1235,14 +1236,23 @@ void Function::adjustAttribute(qreal fraction, int attributeIndex)
         return;
 
     //qDebug() << Q_FUNC_INFO << "idx:" << attributeIndex << ", val:" << fraction;
-    m_attributes[attributeIndex].value = CLAMP(fraction, 0.0, 1.0);
+    if(attributeIndex == Function::CrosfaderId){
+        m_attributes[attributeIndex].value = fraction;
+    } else {
+        m_attributes[attributeIndex].value = CLAMP(fraction, 0.0, 1.0);
+    }
     emit attributeChanged(attributeIndex, m_attributes[attributeIndex].value);
 }
 
 void Function::resetAttributes()
 {
-    for (int i = 0; i < m_attributes.count(); i++)
-        m_attributes[i].value = 1.0;
+    for (int i = 0; i < m_attributes.count(); i++){
+        if(!QString::compare(m_attributes[i].name, "CrosfaderId", Qt::CaseInsensitive)){
```

Comment:
Thanks!
I tested and seems is not the problem. When I quit the "!" I get the error in chaserunner because now it seems I'm putting the intensity to 0 as you said.
    
    FAIL!  : ChaserRunner_Test::adjustIntensity() Compared doubles are not the same (fuzzy compare)
      Actual   (m_scene1->getAttributeValue(Function::Intensity)): 0
      Expected (qreal(1.0))                                      : 1

In the doc it says compare will return 0 when true, so that's why I put it like that:

    [static] int QString::compare(const QString &s1, const QString &s2, Qt::CaseSensitivity cs = Qt::CaseSensitive)
    Compares s1 with s2 and returns an integer less than, equal to, or greater than zero if s1 is less than, equal to, or greater than s2.
    If cs is Qt::CaseSensitive, the comparison is case sensitive; otherwise the comparison is case insensitive.
    Case sensitive comparison is based exclusively on the numeric Unicode values of the characters and is very fast, but is not what a human would expect. Consider sorting user-visible strings with localeAwareCompare().

      int x = QString::compare("aUtO", "AuTo", Qt::CaseInsensitive);  // x == 0
      int y = QString::compare("auto", "Car", Qt::CaseSensitive);     // y > 0
      int z = QString::compare("auto", "Car", Qt::CaseInsensitive);   // z < 0

On the other hand, I saw that when I initialize the CrosfaderId (will change it to two s in next commit) to '0' [in this line](https://github.com/anorod/qlcplus/blob/develop/engine/src/function.cpp#L130), I get the test errors. If I do `registerAttribute(tr("CrosfaderId"));` instead of `registerAttribute(tr("CrosfaderId"), 0);` I pass all the unit tests. Still working on why.